src/lib/classes/access-control-list.js

```diff
@@ -0,0 +1,333 @@
+   */
+  createSolidFile = async (url: String, options: Object = {}) =>
+    solid.fetch(url, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'text/turtle'
+      },
+      ...options
+    });
+
+  /**
+   * @function getParentACL Recursively tries to retrieve the acl file from the parent containers if there is not a direct one
+   * @param {String} url Url of the parent container
+   * @returns {Object} Parent acl fetched file
+   */
+  getParentACL = async (url: String) => {
+    const newURL = new URL(url);
+    const { pathname } = newURL;
+    const hasParent = pathname.length > 1;
+    if (!hasParent) return null;
+    const isContainer = pathname.endsWith('/');
+    let newPathname = isContainer ? pathname.slice(0, pathname.length - 1) : pathname;
+    newPathname = `${newPathname.slice(0, newPathname.lastIndexOf('/'))}/`;
+    const parentURI = `${newURL.origin}${newPathname}`;
+    const result = await solid.fetch(`${parentURI}.acl`);
+    if (result.status === 404) return this.getParentACL(parentURI);
+    if (result.status === 200) return result;
+  };
+
+  /**
+   * @function getACLFile Retrieves the acl file from the network
+   * @returns {Object} Acl fetched file
+   */
+  getACLFile = async () => {
+    try {
+      const result = await solid.fetch(this.aclUri);
+      if (result.status === 403) throw new Error(result.message || 'Not authorized');
+      if (result.status === 404) return this.getParentACL(this.documentUri);
+      return result;
+    } catch (e) {
+      throw e;
+    }
+  };
+
+  /**
+   * @function getSubjects Creates an object based on a ldflex proxy document
+   * @param {Proxy} document The base ldflex proxy document from where the data will be extracted
+   * @returns {Array<Permissions>} A custom object to visualized the acls of a document or container it a Pod
+   */
+  getSubjects = async document => {
+    let subjects = [];
+    for await (const subject of document.subjects) {
+      let agents = [];
+      let modes = [];
+      for await (const agent of subject['acl:agent']) {
+        agents = [...agents, agent.value];
+      }
+      for await (const mode of subject['acl:mode']) {
+        const modeName = mode.value ? mode.value.split('#')[1] : '';
+        modes = [...modes, modeName];
+      }
+      const agentClass = await subject['acl:agentClass'];
+      if (agents.length > 0) subjects = [...subjects, { subject, agents, modes }];
+      if (agentClass) subjects = [...subjects, { subject, agents: null, modes }];
+    }
+    return subjects;
+  };
+
+  /**
+   * @function getPermissions Retrieves the acl file as an array of Permissions objects
+   * @returns {Array<Permissions>} An array of permissions
+   */
+  getPermissions = async () => {
+    try {
+      if (!this.acl) {
+        const file = await this.getACLFile();
+        if (!file) throw new Error('ACL File was not found for the resource');
```

Comment:
Can we internationalize this?

Reply:
@pablo-rodriguez-jd please can you use solidError instead of Error so we can passcode error like 404 so the users can get the code then use it.

Reply:
@james-martin-jd , now, this is using ```throw new SolidError('ACL File was not found for the resource', 'Permission Errors', 404);`` so the user can get the code error and handle it like they want.